packages/cache/src/GenericCache.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Tempest\Cache;
+
+use Closure;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Stringable;
+use Tempest\Cache\Config\CacheConfig;
+use Tempest\DateTime\DateTime;
+use Tempest\DateTime\DateTimeInterface;
+use Tempest\DateTime\Duration;
+use Tempest\Support\Arr;
+use Tempest\Support\Random;
+
+final class GenericCache implements Cache
+{
+    public function __construct(
+        private(set) CacheConfig $cacheConfig,
+        public bool $enabled = true,
+        private ?CacheItemPoolInterface $adapter = null,
+    ) {
+        $this->adapter ??= $this->cacheConfig->createAdapter();
+    }
+
+    public function lock(Stringable|string $key, null|Duration|DateTimeInterface $expiration = null, null|Stringable|string $owner = null): Lock
```

Comment:
I envisioned our cache interface to be pretty lightweight. Do we really need locks?

Reply:
Huh, absolutely! Locking is a very important mechanic to have. If you mean "do we need them in the interface", technically no, however it makes things quite harder. In the cache implementation, we have access to everything we need.

May I ask why do you want the interface to be lightweight? With this PR, this interface should not be expected to be implemented in userland anymore, as opposed to the previous design

Reply:
It's mainly that there are some interfaces in Tempest that feel like they are trying to do A LOT. The container is another example. I don't think it's worth refactoring the container at this point, but that's a case where I mixed up interface and concrete implementation responsibilities. But don't tell Aidan, because he's going to say I should have gone with PSR 😅

I'm mainly wondering whether locking is a method that belongs on the base `Cache` interface, or whether it should go on a separate interface that caches can implement, when they need it. Like eg. `LockableCache`

Reply:
Oh, I see. I personally think it's perfectly fine here, but if you want to move it, it's fine as well as long as the DX stays the same

Reply:
Let's keep it as-is for now then :)